docs/design-documents/20240430-schema-support.md

```diff
@@ -0,0 +1,411 @@
+2. The schema service can be more easily changed.
+
+**Disadvantages**:
+
+1. Makes the tech stack more complex.
+
+#### Option 3: A centralized, external schema registry, accessed by connectors directly
+
+This option is similar to the above, in the sense that a centralized schema
+registry is used. However, in this option, Conduit is not used as an
+intermediary. Rather, **connectors access the schema registry directly**. 
+
+**Advantages**:
+
+1. Shortens the time to implement schema support in Conduit.
+
+**Disadvantages**:
+
+1. Makes the tech stack more complex.
+2. Connectors need to follow the schema service's upgrades more closely.
+
+#### Chosen option
+
+Option 1 keeps the tech stack simple and is in line with Conduit's philosophy of
+being deployed as a single binary. However, without replication, fail-over, etc.
+it cannot be considered production ready.
+
+Options 2 and 3 remove that complexity at the expense of adding a new
+infrastructure item.
+
+Having Conduit as an intermediary makes schema registry updates easier, so
+option 2 is the suggested option.
+
+However, **our goal is to eventually implement option 1**.
+
+### Schema format
+
+This section discusses the schema format to be used.
+
+There are two aspects of this:
+
+- the schema format used internally (when registering a schema in the schema
+  service, updating it, fetching, etc.)
+- the schema format exposed to connector developers (through the Connector SDK)
+
+Having them different makes only sense if we expose our own format in the
+Connector SDK. The internal format is dictated by the schema registry that we
+will use, which supports only widely known formats, i.e. we won't be able to use
+our own schema format for that purpose.
+
+**Advantages**:
+1. We can decouple the Connector SDK and Conduit release cycle from the schema
+   internal format release cycle
+2. We want to limit or add feature on top of internal the schema format
+3. We can more easily switch the internal schema format, if needed
+
+**Disadvantages**:
+
+1. Newer features and fixes in the schema format used internally (e.g. Avro)
+   sometimes need to be explicitly added to the schema format used
+2. Boilerplate code that converts the SDK schema into the internal schema
+3. Boilerplate code that exposes schema-related operations (encode/decode record payloads)
+
+#### Internal schema format
+
+##### Option 1: Avro
+
+We use Avro as the schema format used by the Connector SDK and internally.
+
+**Advantages**:
+
+1. Schema is a first-class citizen
+2. A widely used schema format.
+3. A popular option with Kafka Connect (makes it easier for users to migrate)
+
+**Disadvantages**:
+
+##### Option 2: Protobuf schema
+
+**Advantages**
+
+1. Faster (de)serialization
+
+**Disadvantages**:
+
+1. Protobuf libs don't offer a way to create a `.proto` file, i.e. that needs to
+   be done manually.
+
+#### Chosen option
+
+
+
+### Schema service interface
```

Comment:
Explore possibility of "reversing" a connection (Conduit initiates connection to connector, then connector establishes a connection to Conduit)

Reply:
@hariso is this required for this release? Wondering if that can be tackled independently or should be part of this design document.

Reply:
I think I can do that as part of the SDK work, where I'll need that connection from a connector to Conduit.